python-package/xgboost/dask/__init__.py

```diff
@@ -1444,6 +1445,18 @@ def _dispatch(ref: Optional[DaskDMatrix], **kwargs: Any) -> DaskDMatrix:
     return train_dmatrix, awaited
 
 
+def _mapped_predict(
+    partition: DataFrame | numpy.ndarray,
```

Comment:
This type signature isn't quite right. This could also get a `cupy.ndarray`, depending on the data in the dask array.

Reply:
Looks like `DataType` (which is an alias for `Any` at the moment) is what we should use.